core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -74,6 +76,34 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
   def recoverWith[A](fa: F[A])(pf: PartialFunction[E, F[A]]): F[A] =
     handleErrorWith(fa)(e =>
       pf applyOrElse(e, raiseError))
+  /**
+   * Often E is Throwable. Here we try to call pure or catch
+   * and raise.
+   */
+  def catchNonFatal[A](a: => A)(implicit ev: Throwable =:= E): F[A] =
```

Comment:
My first inclination would be to make this `eq: Throwable <:< E`—that will work in all reasonable places where `E` is statically known, and it eliminates the need for guesswork about whether to write `E =:= Throwable` (which won't work) or `Throwable =:-=` (which will) for people who want to support these methods in generic contexts.